src/pages/søknad/steg/bo-og-opphold-i-norge/Bo-og-opphold-i-norge.tsx

```diff
@@ -100,10 +110,26 @@ const schema = yup.object<FormData>({
             },
         }),
     fortsattInnlagt: yup.boolean(),
+    borPåAdresse: yup
+        .mixed<Adresse>()
+        .nullable()
+        .test({
+            name: 'borPåAdresse',
+            message: 'En av borPåAdresse eller ingenAdresseGrunn må vare satt',
+            test: function () {
+                return Boolean(this.parent.borPåAdresse || this.parent.ingenAdresseGrunn);
+            },
+        }),
+    ingenAdresseGrunn: yup.mixed<IngenAdresseGrunn>().nullable(),
 });
 
 const BoOgOppholdINorge = (props: { forrigeUrl: string; nesteUrl: string }) => {
-    const boOgOppholdFraStore = useAppSelector((s) => s.soknad.boOgOpphold);
+    const {
+        søker: { søker },
+        soknad,
+    } = useAppSelector((s) => s);
```

Comment:
Hvis det gjøres på denne måten så vil denne komponenten re-rendre for hver eneste endring i `store`. Dersom vi heller plukker ut akkurat det vi er interesserte i vil `react-redux` passe på at vi kun re-rendrer dersom noen av feltene vi trenger oppdateres.

```suggestion
    const { søker, soknad } = useAppSelector((s) => ({
        søker: s.søker.søker,
        soknad: s.soknad
    }));
```

Har ikke sett igjennom hele komponenten nå, men dersom det er bare `soknad.boOgOpphold` som brukes, så kunne det vært fordelaktig å kun hente ut den verdien fra `store` også.

Reply:
Aaah, cool! 👍